### DISCUSSION TITLE
Enhancing Pathfinder Documentation

### DISCUSSION THREAD

**Discussion by TropicolX** [eqlabs/pathfinder#2031] [Ideas]
Hi @Mirko-von-Leipzig and Pathfinder Team,

I hope you're doing well! My name is Oluwabusayo, and I'm a technical writer who loves diving into open-source projects. I recently came across Pathfinder and was really impressed by what it offers as a node client for Starknet. However, I noticed the documentation site is still a work in progress:
![image](https://github.com/eqlabs/pathfinder/assets/68024640/a6dccefd-1588-4a4d-91f0-2ab911bcaf22)

I’d love to help out by creating a comprehensive documentation site. Right now, there’s just the GitHub README, and I think we can do a lot more to make Pathfinder more user-friendly and accessible.

Here’s what I’m thinking:
- **Detailed Documentation**: Extract and expand the current Github README into a full documentation site using [Docusaurus](https://docusaurus.io/) with clear sections on installation, setup, and usage. This would be similar to [Starknet's official documentation](https://docs.starknet.io/documentation/) or competitors' docs like [Nethermind](https://docs.nethermind.io/) and [Juno](https://juno.nethermind.io/).
![image](https://github.com/eqlabs/pathfinder/assets/68024640/0e5778bb-7b25-40c4-93a1-2807ea8d2f08)

- **User Guides and Tutorials**: Step-by-step guides to help new users get started quickly.
- **API References**: Detailed and updated documentation of the Starknet JSON-RPC API and the Monitoring API so developers can easily integrate them.
- **FAQ**: A helpful section to address common issues and frequently asked questions.

This would be an open-source contribution, so I do not require any compensation. All I’d need to do is connect with some team members to gain more insights into the tool, understand common issues, and gather the necessary information to draft the main sections of the documentation.

If you’re interested, I’d love to chat more about how we can get started.

**Comment by Mirko-von-Leipzig** [comment]
Hi @TropicolX - honestly your suggestion sounds great to us. We've had plans for more formal documentation but as you can tell we've never gotten to actually implementing them.

I can definitely dedicate some time to get the ball rolling on this. What would be easiest for you ito communication channel? We can have a short introductory video call, or we can keep things async here on github or some other tool if you prefer.

**Reply by TropicolX** [reply]
Hi @Mirko-von-Leipzig 
Thanks for the positive feedback! Discord or Slack would work great for me for communication. Let me know which one works better for you. We can keep things relaxed and flexible there and manage everything asynchronously for now.

Looking forward to collaborating!

**Reply by Mirko-von-Leipzig** [reply]
Awesome, will reach out via email to set that up :) Chat soon
